pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -446,6 +498,31 @@ func (p *PodAutoscalerInternal) BuildStatus(currentTime metav1.Time, currentStat
 	return status
 }
 
+// Private helpers
+func addHorizontalAction(currentTime time.Time, retention time.Duration, actions []datadoghq.DatadogPodAutoscalerHorizontalAction, action *datadoghq.DatadogPodAutoscalerHorizontalAction) []datadoghq.DatadogPodAutoscalerHorizontalAction {
+	if retention == 0 {
+		actions = actions[:0]
+		actions = append(actions, *action)
+		return actions
```

Comment:
```suggestion
		return actions[:0]
```
I think it's more concise. If the goal was to update `actions` in place. I am not sure that it works. Example: https://go.dev/play/p/_iaqW7Zwg3k